django/contrib/admin/widgets.py

```diff
@@ -450,6 +450,20 @@ def __init__(self, attrs=None):
 SELECT2_TRANSLATIONS.update({"zh-hans": "zh-CN", "zh-hant": "zh-TW"})
 
 
+def get_select2_language():
+    lang_code = get_language()
+    possible_lang_codes = [lang_code]
+    i = None
+    while (i := lang_code.rfind("-", 0, i)) > -1:
+        possible_lang_codes.append(lang_code[:i])
```

Comment:
We can immediately return a supported language instead of building a temporary list.